hpx/runtime/components/client_base.hpp

```diff
@@ -289,11 +289,18 @@ namespace hpx { namespace components
         }
 
         ///////////////////////////////////////////////////////////////////////
+#if defined(HPX_HAVE_COMPONENT_GET_GID_COMPATIBILITY)
         naming::id_type const & get_gid() const
         {
             return gid_.get();
         }
 
+#endif
+        naming::id_type const & get_id() const
+        {
+            return gid_.get();
+        }
+
```

Comment:
Hmm, the id stored in a client could be both, managed or unmanaged, we don't know this. Does the user ever want to retrieve an id which is guaranteed to be unmanaged? Do you have an use case?